### ISSUE TITLE
Plugin doesn't work wit exciting env and new pipeline 

### ISSUE THREAD

**Issue by 99** [tomzo/gocd-json-config-plugin#4]
I've installed the plugin, created the *.gopipeline.json and was trying to add the plugin to our goCd server. But it seems not working for our case.
The reason is we're using environments, and there is no way of specifying the exciting (_default_) environment for a newly created pipeline using the current plugin. 

I tried to add pipeline name to the cruise-config.xml file but got an error that pipeline doesn't exist which make sense. 
Can any workaround be suggested, please?

**Comment by tomzo** [comment]
When you add a pipeline in json or yaml, let's call it `remote-pipeline`, using [config repo feature](https://docs.gocd.io/current/advanced_usage/pipelines_as_code.html) you will not be able to configure that this pipeline belongs to environment via UI or directly in cruise-config.xml. You must define the relationship in some configuration repository. E.g. Adding file like this to your current repository would solve it:

`myenv.goenvironment.json`
```json
{
  "name": "myenv",
  "pipelines": [
    "remote-pipeline"
  ]
}
```

If you have `myenv` already in cruise-config.xml, that's OK, it will be merged with the main configuration.

**Comment by 99** [comment]
Thank you @tomzo  for the quick response, but the pipeline is not running.
It's in the weird state (Unknown) it's not active ( I can't run/pause it).
And env settings displaying this : 

<img width="401" alt="default_-_environment_-_go" src="https://cloud.githubusercontent.com/assets/5048814/23443232/3bbbabe4-fde2-11e6-92d3-a61f5665fe27.png">

**Comment by tomzo** [comment]
@nserebry  If the pipeline does not schedule then it might have misconfigured materials. It hard to tell for me what you did wrong. But once pipeline is accepted by GoCD as valid, it's behavior is the same as if configured in xml. The env settings look as expected to me.

**Comment by 99** [comment]
I found the issue. 
If I'm using *.goenvironment.json and after trying to modify (add/delete another pipeline)  for that environment  using UI, it will return error 'null' (_nothing n logs_)
Example : if my myenv.goenvironment.json file is :
```
{
  "name": "myenv",
  "pipelines": [
    "remote-pipeline"
  ]
}

```

And then I created a new pipeline and want to add it by going to UI - > 
 Environment - "myenv" -> Edit 
checkbox a new pipleine
click "Save"
see error "null"
<img width="687" alt="default_-_environment_-_go" src="https://cloud.githubusercontent.com/assets/5048814/23484283/3d92a856-fe8b-11e6-9b70-6bda12a5e829.png">

**Comment by tomzo** [comment]
I think this might be issue causing it https://github.com/gocd/gocd/issues/2943

**Comment by 99** [comment]
@tomzo sounds similar, but I'm getting nothing in the go-server logs,  except this message, felt liek config was modified. UI failing with 'null' error.
```
2017-03-02 00:24:55,425  INFO [qtp2143192188-2282962] GoConfigDao:91 - Config update for pipeline request by com.thoughtworks.go.server.domain.Username@128759ea[displayName=XXX,username=XXX] is in queue - com.thoughtworks.go.config.update.UpdateEnvironmentCommand@143c7a2d
2017-03-02 00:24:55,426  INFO [qtp2143192188-2282962] GoConfigDao:93 - Config update for pipeline request by com.thoughtworks.go.server.domain.Username@128759ea[displayName=XXX,username=XXX] is being processed

```

**Comment by tomzo** [comment]
fixed in server 17.3.0
